docs/tables/aws_alb_access_log/queries.md

```diff
@@ -0,0 +1,347 @@
+  target_processing_time,   -- Time taken by target to process request
+  response_processing_time, -- Time taken to process response
+  (request_processing_time + target_processing_time + response_processing_time) as total_time
+from
+  aws_alb_access_log
+where
+  (request_processing_time + target_processing_time + response_processing_time) > 1 -- Requests taking longer than 1 second
+order by
+  total_time desc
+limit 10;
+```
+
+### Target health issues
+
+Identify targets that are returning a high number of errors.
+
+```sql
+select
+  target_ip,
+  target_status_code,
+  count(*) as error_count
+from
+  aws_alb_access_log
+where
+  target_status_code >= 400
+group by
+  target_ip,
+  target_status_code
+having
+  count(*) > 100
+order by
+  error_count desc;
+```
+
+## Volume Examples
+
+### High traffic periods
+
+Detect periods of unusually high request volume.
+
+```sql
+select
+  date_trunc('minute', timestamp) as request_minute,
+  elb,
+  count(*) as request_count
+from
+  aws_alb_access_log
+group by
+  request_minute,
+  elb
+having
+  count(*) > 1000
+order by
+  request_count desc;
+```
+
+### Large response sizes
+
+Track requests generating unusually large responses.
+
+```sql
+select
+  timestamp,
+  elb,
+  tp_index as account_id,
+  request,
+  client_ip,
+  sent_bytes,
+  received_bytes
+from
+  aws_alb_access_log
+where
+  sent_bytes > 10485760 -- 10MB
+order by
+  sent_bytes desc;
+```
+
+### Count requests by Listener Type (HTTP, HTTPS, HTTP/2, WebSockets, etc.)
+
+Identify how many requests were received for each listener type.
+
+```sql
+select
+  listener_type,
+  count(*) as request_count
+from (
+  select
+    case
+      when request like 'GET http://%' then 'HTTP'
+      when request like 'GET https://%' and ssl_protocol is not null then 'HTTPS'
+      when request like 'GET https://%' and ssl_protocol is null then 'HTTP/2'
+      when request like 'GET ws://%' then 'WebSockets'
+      when request like 'GET wss://%' then 'Secured WebSockets'
+      else 'Other'
+    end as listener_type
+  from aws_alb_access_log
+)
+group by listener_type
+order by request_count desc;
+```
+
+### Requests with Invalid Cookies
+
+Identify requests with invalid cookies.
+
+```sql
+select
+  timestamp,
+  client_ip,
+  request,
+  error_reason
+from
+  aws_alb_access_log
+where
+  error_reason = 'AuthInvalidCookie'
+order by
+  timestamp desc;
+```
+
+### Requests with Invalid Lambda response
+
+Identify requests with an invalid Lambda response.
+
+```sql
+select
+  timestamp,
+  client_ip,
+  request,
+  error_reason
+from
+  aws_alb_access_log
+where
+  elb_status_code = 502
+  and error_reason like 'LambdaInvalidResponse'
+order by
+  timestamp desc;
+```
+
+### Requests blocked by WAF rules
+
+Identify requests blocked by WAF rules.
+
+```sql
+select
+  timestamp,
+  client_ip,
+  request,
+  error_reason
+from
+  aws_alb_access_log
+where
+  actions_executed = 'waf'
```

Comment:
What type is `actions_executed`? According to https://docs.aws.amazon.com/elasticloadbalancing/latest/application/load-balancer-access-logs.html#access-log-entry-syntax, it's a comma-separated list of actions from https://docs.aws.amazon.com/elasticloadbalancing/latest/application/load-balancer-access-logs.html#actions-taken, so should our column also be a list of strings?